src/clients/api/queries/useUserMarketInfo.ts

```diff
@@ -58,71 +63,130 @@ const useUserMarketInfo = ({
-        isEnabled = toDecimalAmount(wallet.tokenAllowance).isGreaterThan(walletBalance);
+  const {
+    assets,
+    userTotalBorrowBalanceCents,
+    userTotalBorrowLimitCents,
+    userTotalSupplyBalanceCents,
+    treasuryTotalUsdBalanceCents,
+    treasuryTotalSupplyUsdBalanceCents,
+    treasuryTotalBorrowUsdBalanceCents,
+    treasuryTotalAvailableLiquidityUsdBalanceCents,
+  } = Object.values(TOKENS).reduce(
+    (acc, item, index) => {
+      const { assets: assetAcc } = acc;
+      const toDecimalAmount = (mantissa: string) =>
+        new BigNumber(mantissa).shiftedBy(-item.decimals);
+      const vBepToken = getVBepToken(item.id);
+      // if no corresponding vassets, skip
+      if (!vBepToken) {
+        return acc;
+      }
+
+      const market = marketsMap[item.id] || {};
+      const vtokenAddress = vBepToken.address.toLowerCase();
+      const collateral = assetsInAccount
+        .map((address: string) => address.toLowerCase())
+        .includes(vtokenAddress);
+
+      let treasuryBalance = new BigNumber(0);
+      if (treasuryBalances[vtokenAddress]) {
+        treasuryBalance = toDecimalAmount(treasuryBalances[vtokenAddress].tokenBalance);
+      }
+
+      let walletBalance = new BigNumber(0);
+      let supplyBalance = new BigNumber(0);
+      let borrowBalance = new BigNumber(0);
+      let isEnabled = false;
+      const percentOfLimit = '0';
+
+      const wallet = balances[vtokenAddress];
+      if (accountAddress && wallet) {
+        walletBalance = toDecimalAmount(wallet.tokenBalance);
+        supplyBalance = toDecimalAmount(wallet.balanceOfUnderlying);
+        borrowBalance = toDecimalAmount(wallet.borrowBalanceCurrent);
+        if (item.id === 'bnb') {
+          isEnabled = true;
+        } else {
+          isEnabled = toDecimalAmount(wallet.tokenAllowance).isGreaterThan(walletBalance);
+        }
+      }
+
+      const asset = {
+        key: index,
+        id: item.id,
+        img: item.asset,
+        vimg: item.vasset,
+        symbol: market.underlyingSymbol || item.id.toUpperCase(),
+        decimals: item.decimals,
+        tokenAddress: market.underlyingAddress,
+        vsymbol: market.symbol,
+        vtokenAddress,
+        supplyApy: new BigNumber(market.supplyApy || 0),
+        borrowApy: new BigNumber(market.borrowApy || 0),
+        xvsSupplyApy: new BigNumber(market.supplyVenusApy || 0),
+        xvsBorrowApy: new BigNumber(market.borrowVenusApy || 0),
+        collateralFactor: new BigNumber(market.collateralFactor || 0).div(1e18),
+        tokenPrice: new BigNumber(market.tokenPrice || 0),
+        liquidity: new BigNumber(market.liquidity || 0),
+        borrowCaps: new BigNumber(market.borrowCaps || 0),
+        treasuryTotalBorrowsUsd: new BigNumber(market.totalBorrowsUsd),
+        treasuryTotalSupplyUsd: new BigNumber(market.totalSupplyUsd),
+        treasuryTotalSupply: new BigNumber(market.totalSupply),
+        treasuryTotalBorrows: new BigNumber(market.totalBorrows2),
```

Comment:
I should have made this obvious, but here too we should follow our patterns of adding the unit as suffix